src/sentry/notifications/notifications/integration_nudge.py

```diff
@@ -0,0 +1,108 @@
+
+from sentry.db.models import Model
+from sentry.notifications.notifications.base import BaseNotification
+from sentry.notifications.utils.actions import MessageAction
+from sentry.types.integrations import ExternalProviders
+
+if TYPE_CHECKING:
+    from sentry.models import Organization, Team, User
+
+logger = logging.getLogger(__name__)
+
+MESSAGE_LIBRARY = [
+    "Check your email lately? We didn't think so. Get Sentry notifications in {provider}.",
+    "Complicated relationship with email? We understand. We can notify you in {provider} instead.",
+    "Dread email? We get it. Hear from Sentry in {provider} instead.",
+    "Emails are so 2010s. Get Sentry notifications in {provider}.",
+    "Everyone has feelings about notifications. Whatever yours are, get your"
+    " needs met with Sentry for {provider}.",
+    "Get notifications where you'll actually see them. Hear from Sentry in {provider}.",
+    "Is this a dream come true? Or notifications out of control? Hear from"
+    " Sentry right here in {provider}. If you want.",
+    "Like getting notified in {provider}? Sentry can do that. Want to be left alone? Ignore this.",
+    "Like this notification? There's more where that came from. Hear from Sentry right here in {provider}.",
+    "Notifications aren't always bad. Hear from Sentry right here in {provider}.",
+    "Remember when email was exciting? It's not 1998 anymore. Get notified in {provider} instead.",
+    "Reminisce on the days before your email address was an obligational burden"
+    " while signing up for {provider} notifications instead.",
+    "This is a notification about notifications. Now you can hear from Sentry right here in {provider}.",
+    "This isn't 1998. Swap emails for Sentry notifications in {provider}.",
+    "We're doing something new. Get Sentry notifications right here in {provider}.",
+    "We're here to notify you about notifications. Meta, huh? Hear from Sentry right here in {provider}.",
+    "We've got something new for you. Get Sentry notifications right here in {provider}.",
+    "Who even checks their email anymore? Get Sentry notifications in {provider}.",
+    "You've got mail. So much mail. You're lost and confused as you cut through"
+    " a jungle of mail. Sentry can notify you in {provider} instead.",
+]
+
+
+class IntegrationNudgeNotification(BaseNotification):
+    category = "integration_nudge"
+    filename = "integration-nudge"
+    type = "integration.nudge"
+
+    def __init__(
+        self,
+        organization: Organization,
+        recipient: User,
+        provider: ExternalProviders,
+        seed: int | None = None,
+    ) -> None:
+        super().__init__(organization)
+        self.recipient = recipient
+        self.provider = provider
+        self.seed = (
+            (seed % len(MESSAGE_LIBRARY))
+            if seed is not None
+            else random.randint(0, len(MESSAGE_LIBRARY) - 1)
+        )
+
+    def get_reference(self) -> Model | None:
```

Comment:
Are most of these methods just required from the base class? Is that why they just return empty values?

Reply:
Yeah they're used in constructing the email version of the message.

Reply:
@mgaeta right...but why do we need to define them if we aren't sending an email?